model/project_ref.go

```diff
@@ -1106,19 +1110,38 @@ func GetTasksWithOptions(projectName string, taskName string, opts GetProjectTas
 		task.DisplayNameKey: taskName,
 		task.StatusKey:      bson.M{"$in": finishedStatuses},
 	}
+	var repo *Repository
+	var revisionMatch []bson.M
+	if opts.NumVersions > 0 {
```

Comment:
Perhaps rather than allowing this to be user preferenced, we can just start enforcing this with a large number (1000 versions or something).

Reply:
SGTM